src/main.cc

```diff
@@ -0,0 +1,88 @@
+#ifdef ONPC
+#include "../algo/debug.h"
+#else
+#define debug(...) 42
+#endif
+#include <cassert>
+#include <cstdint>
+#include <utility>
+#include <vector>
+
+template<typename T>
+using Pair = std::pair<T, T>;
+
+template<typename T, typename Functor>
+struct FunctorQueue {
+private:
+  std::vector<Pair<T>> stack_first, stack_second;
```

Comment:
Btw, you limited your code possibilities by coupling value and dynamics into one Pair structure. 
E.g. we want to make `emplace(Args&&...)` method. If value and dynamics were decoupled , we would pass args to value stack and create object there, and then pass it by reference to functor. In your solution it is not possible.